account_invoice_change_currency/models/res_currency.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2017-2018 Vauxoo
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class ResCurrency(models.Model):
+    _inherit = "res.currency"
+
+    def _get_rates(self, company, date):
+        currency_rates = super()._get_rates(company, date)
+        custom_rate = self.env.context.get("custom_rate")
+        to_currency = self.env.context.get("to_currency")
+        if custom_rate and to_currency:
+            currency_rates[to_currency.id] = custom_rate
```

Comment:
Not enough I think, because you need to provide rate for the other currency explicitly, so you ensure it's 1.0. Otherwise, other one could be retrieved by Odoo